### PR TITLE
Fixes #RHIROS-968 - Added DB migration scripts

### PR DESCRIPTION
**Pull request by patilsuraj767**

### CODE REVIEW

File: cmd/db.go [hunk 1]
```diff
@@ -2,29 +2,57 @@ package cmd
...
+var migratedown = &cobra.Command{
+	Use:   "down",
+	Short: "Reverse database migration",
+	Long:  "Reverse database migration",
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("Reverse database migration")
+		// Placeholder for db downgrade.
+		// This will be helpful for force unlock dirty migration
 	},
 }
```
Comment (saltgen) [inline cmd/db.go:57]: A couple of command suggestions, I think `head` will be especially useful.

- `revisions` (a list of all applied migrations)
- `head` (details about the latest applied migration)

Comment (patilsuraj767) [reply cmd/db.go:62]: I have added only one sub-command `revision` to get the migration details. 
`golang-migrate` scripts does not add much to `schema_migrations` table to display. 
```
postgres=# select * from schema_migrations ;
 version | dirty 
---------+-------
       4 | f
(1 row)
```

Comment (saltgen) [reply cmd/db.go:62]: Sounds good, as long as we can make a bit of sense about applied/pending migrations :+1:

Comment (saltgen) [reply cmd/db.go:62]: IIUC the `revision` ~ `schema_migrations` should let me know how many migrations are applied atm.
So I went with `down` a few times but the revisions show me 4 everytime, am I missing something?

```
( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db migrate up                                                                             (kind-kind) (ros-ocp-backend) 15:36:11
Forward database migration
no change
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db migrate down                                                                           (kind-kind) (ros-ocp-backend) 15:36:49
Reverse database migration
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db revision                                                                               (kind-kind) (ros-ocp-backend) 15:36:54
Current migration version is: 4 
Is dirty: false 
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db migrate down                                                                           (kind-kind) (ros-ocp-backend) 15:36:58
Reverse database migration
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db revision                                                                               (kind-kind) (ros-ocp-backend) 15:37:02
Current migration version is: 4 
Is dirty: false 
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db migrate down                                                                           (kind-kind) (ros-ocp-backend) 15:37:05
Reverse database migration
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db revision                                                                               (kind-kind) (ros-ocp-backend) 15:37:32
Current migration version is: 4 
Is dirty: false 
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db migrate down                                                                           (kind-kind) (ros-ocp-backend) 15:37:34
Reverse database migration
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db revision                                                                               (kind-kind) (ros-ocp-backend) 15:37:36
Current migration version is: 4 
Is dirty: false 

```

File: migrations/000001_create_rh_accounts_table.up.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE IF NOT EXISTS rh_accounts(
```
Comment (saltgen) [inline migrations/000001_create_rh_accounts_table.up.sql:1]: Is there a way we can contain the `up` and `down` in a single file in `migrations/`?

File: cmd/db.go (hunk 1 above)
Comment (patilsuraj767) [reply cmd/db.go:62]: I haven't implemented `down` subcommand. I don't want that someone by mistakenly run `down` command in production DB. 
But to test it. you can run `make install-golang-migrate-cli-tool` to install migrate cli tool and then run below command. 

`./bin/migrate -path ./migrations/ -database "postgres://postgres:postgres@localhost:15432/postgres?sslmode=disable" down
`

File: migrations/000001_create_rh_accounts_table.up.sql (hunk 2 above)
Comment (patilsuraj767) [reply migrations/000001_create_rh_accounts_table.up.sql:1]: No, we can't. This is how [golang -migrate](https://github.com/golang-migrate/migrate) is design to work.

File: cmd/db.go (hunk 1 above)
Comment (saltgen) [reply cmd/db.go:62]: IMO, we need both `up` and `down`, IIUC our PROD DB would be on RDS so the concerned team can manage it.
If not, we can have appropriate documentation later about these.

Comment (patilsuraj767) [reply cmd/db.go:87]: I added the `down` command.

**Review by saltgen — APPROVED** [review]
Thanks @patilsuraj767, changes look good!

**Pull request merged**